Assets/Plugins/LeapMotion/Core/Editor/LeapServiceProviderEditor.cs

```diff
@@ -77,63 +97,641 @@ public class LeapServiceProviderEditor : CustomEditorBase<LeapServiceProvider> {
+
+      if (LeapController?.Devices?.Count == 1) {
+        if (LeapController.Devices.First().Type == Device.DeviceType.TYPE_RIGEL) {
+          DrawRigelInteractionZoneMesh();
+        } else if (LeapController.Devices.First().Type == Device.DeviceType.TYPE_PERIPHERAL) {
+          DrawLeapMotionControllerInteractionZone(LMC_BOX_WIDTH, LMC_BOX_DEPTH, LMC_BOX_RADIUS, Color.white);
+        }
+      }
+    }
+
+    private LeapServiceProvider.InteractionVolumeVisualization? GetSelectedInteractionVolume() {
+
+      return LeapServiceProvider?.SelectedInteractionVolumeVisualization;
+    }
+
+
+    private void DrawRigelInteractionZoneMesh() {
 
-      Handles.DrawLine(origin, top_left);
-      Handles.DrawLine(origin, top_right);
-      Handles.DrawLine(origin, bottom_left);
-      Handles.DrawLine(origin, bottom_right);
+      if (this.rigelInteractionZoneMesh != null && this.rigelInteractionZoneMesh.Edges() != null) {
+        foreach (Edge edge in this.rigelInteractionZoneMesh.Edges()) {
+          
+          // Draw edges
+          if (edge.CommonNormal != null) {
+            Vector3 edgeVertex0 = target.transform.TransformPoint(edge.VertexLocations[0].Item2);
+            Vector3 edgeVertex1 = target.transform.TransformPoint(edge.VertexLocations[1].Item2);
+            Vector3 edgeCommonNormal = target.transform.TransformPoint(edge.CommonNormal);
 
-      drawControllerEdge(origin, local_top_left, local_top_right);
-      drawControllerEdge(origin, local_bottom_left, local_top_left);
-      drawControllerEdge(origin, local_bottom_left, local_bottom_right);
-      drawControllerEdge(origin, local_bottom_right, local_top_right);
+            float angle;
+            if (SceneView.currentDrawingSceneView.camera.orthographic) {
+              // Iso Camera
+              angle = Vector3.Angle(SceneView.currentDrawingSceneView.camera.transform.forward, edgeCommonNormal);
+            } else {
+              // Perspective Camera
+              angle = Vector3.Angle(edgeVertex0 - SceneView.currentDrawingSceneView.camera.transform.position, edgeCommonNormal);
+            }
+
+            // Shade the edge line based on it's angle to the camera, making it more intense as it approaches 90 degrees
+            // This gives the interaction volume a cell style appearance
+            float shadingFactor = Math.Abs(angle / 90.0f);
+            if (shadingFactor > 1) {
+              shadingFactor = 2 - shadingFactor;
+            }
+
+            if (angle >= Math.Abs(90)) // Was <= if YZ flipping is not on 
+            {
+              Handles.color = new Color(shadingFactor, shadingFactor, shadingFactor, shadingFactor * PEAK_INTERACTION_VOLUME_OPACITY);
+              Handles.DrawAAPolyLine(edgeVertex0, edgeVertex1);
+            } else // Edge is effectively a backface edge
+            {
+              Handles.color = BackfaceEdgeColour;
+              Handles.DrawAAPolyLine(edgeVertex0, edgeVertex1);
+            }
```

Comment:
This is a fun way to achieve the "cell-style" appearance, but there's an unfortunate side-effect to drawing lines one at a time this way:
![image](https://user-images.githubusercontent.com/174475/90080841-8a479500-dcc0-11ea-9298-670178cc5cc8.png)
It's horrifically slow!   The editor slows down to a crawl when the wireframe is displayed.

When `DrawAAPolyLine()` is called with only two arguments, it enqueues a single drawcall to the command buffer to execute drawing.  This is fine if it's only a single digit number of line segments, but we've got 4671 edges (and thus 4671 draw calls) being enqueued!

The easy fix is to queue up your edges into a giant Vector3[] and [call `DrawAAPolyLine()` with that array as an input argument.](https://docs.unity3d.com/ScriptReference/Handles.DrawAAPolyLine.html)  This will draw all of the lines as a single draw call.

The side effect of this is that you won't be able to change the color of each line segment at a granular level.  

The proper fix is to draw the cage with [`Graphics.DrawMeshNow()`](https://docs.unity3d.com/ScriptReference/Graphics.DrawMeshNow.html) and to use a Shader+Material to create the Transparency look that you're going for.

So you'd first create the material, and then you'd replace this whole edge iteration loop with something like:
```
// On Initialization
      rigelInteractionMaterial = (Material)AssetDatabase.LoadAssetAtPath(Path.Combine("Assets", "Plugins", "LeapMotion", "Core", "Materials", "Default.mat"), typeof(Material));
...
// Then later inside of DrawRigelInteractionZoneMesh()
      rigelInteractionMaterial.SetPass(0);
      Graphics.DrawMeshNow(rigelInteractionMesh, Matrix4x4.TRS(Vector3.zero, Quaternion.Euler(-90,0,0), Vector3.one*0.001f));
```